### PR TITLE
documenting auth mechanisms

### PR DESCRIPTION
**Pull request by anaxceron**
Adding documentation for authentication mechanisms:
- New section in [Developing for Zowe CLI](https://docs.zowe.org/stable/extend/extend-cli/cli-devTutorials)
- New page in the [Troubleshooting Zowe CLI](https://docs.zowe.org/stable/troubleshoot/cli/troubleshoot-cli)
- updated sidebar to include the new Troubleshoot page

### CODE REVIEW

File: docs/extend/extend-cli/cli-devTutorials.md [hunk 1]
```diff
@@ -8,13 +8,27 @@ You can contribute to Zowe CLI in the following ways:
 - Develop a plug-in that users can install to Zowe CLI.
```
Comment (sbohnert) [inline docs/extend/extend-cli/cli-devTutorials.md:8]: Sounds odd. How about "Develop a plug-in to contribute to Zowe CLI"?

Comment (adam-wolfe) [reply docs/extend/extend-cli/cli-devTutorials.md:8]: It does sound weird.

Maybe it could be simplified to: "Develop a Zowe CLI plug-in".

**Review by zFernand0 — APPROVED** [review]
I really like how this turned out!
LGTM! 😋

**Review by traeok — APPROVED** [review]
LGTM, thanks @anaxceron!

**Pull request merged**